### ISSUE TITLE
Inlcude GET headers in dataSet object

### ISSUE THREAD

**Issue by gibsonf1** [inrupt/solid-client-js#779]
It would be great if the dataset could harvest the headers as an object, maybe dataset.headers.  Our particular case is we need to check the eTag in the header to see if a resource has been updates since last load, and then we wouldn't need to do an additional HEAD after loading the dataset to get that header.

**Comment by nicolasmondada** [comment]
Hi @gibsonf1 ,

What you are asking for is a way to check if a resource that is in-memory is up-to-date with the server or needs to be re-fetched (optionally just re-fetch it if it's not up-to-date in the same check call).

Do I understand you correctly?

Cheers,
Nicolas.-

**Comment by gibsonf1** [comment]
@nicolasmondada   Yes, we have built a system around eTag to know when to refetch a resource thats changed.
